tests/integration/targets/nd_flow_rules/tasks/main.yml

```diff
@@ -0,0 +1,175 @@
+      - nm_add_flow_rule_vrf_2.current.tenant == "ansible_tenant_test"
+      - nm_add_flow_rule_vrf_2.current.vrf == "ansible_vrf_test"
+      - nm_add_flow_rule_vrf_2.current.flowRuleAttributeList == []
+
+# UPDATE TASKS
+- name: Update current subnets for VRF Flow Rule
+  cisco.nd.nd_flow_rules: &ndi_flow_rule_vrf_update
+    <<: *ndi_flow_rule_vrf_present
+    subnets:
+      - 10.10.1.0/24
+      - 10.10.2.0/24
+  register: nm_update_subnets_flow_rule_vrf
+
+- name: Update current subnets for VRF Flow Rule - again
+  cisco.nd.nd_flow_rules:
+    <<: *ndi_flow_rule_vrf_update
+  register: nm_update_again_subnets_flow_rule_vrf
+
+- name: Delete all existing subnets for VRF Flow Rule
+  cisco.nd.nd_flow_rules:
+    <<: *ndi_flow_rule_vrf_update
+    subnets: []
+  register: nm_delete_subnets_flow_rule_vrf
+
+- name: Asserts for VRF Flow Rules update tasks
+  ansible.builtin.assert:
+    that:
+      - nm_update_subnets_flow_rule_vrf is changed
+      - nm_update_subnets_flow_rule_vrf.data.flowRuleAttributeList.0.subnet == "10.10.0.0/24"
+      - nm_update_subnets_flow_rule_vrf.data.flowRuleAttributeList.0.operation == "DELETE"
+      - nm_update_subnets_flow_rule_vrf.data.flowRuleAttributeList.1.subnet == "10.10.2.0/24"
+      - nm_update_subnets_flow_rule_vrf.data.flowRuleAttributeList.1.operation == "ADD"
+      - nm_update_subnets_flow_rule_vrf.current.name == "ansible_test"
+      - nm_update_subnets_flow_rule_vrf.current.tenant == "test"
+      - nm_update_subnets_flow_rule_vrf.current.vrf == "test"
+      - nm_update_subnets_flow_rule_vrf.current.flowRuleAttributeList.0.subnet == "10.10.1.0/24"
+      - nm_update_subnets_flow_rule_vrf.current.flowRuleAttributeList.1.subnet == "10.10.2.0/24"
+      - nm_update_again_subnets_flow_rule_vrf is not changed
+      - nm_update_again_subnets_flow_rule_vrf.current.name == "ansible_test"
+      - nm_update_again_subnets_flow_rule_vrf.current.tenant == "test"
+      - nm_update_again_subnets_flow_rule_vrf.current.vrf == "test"
+      - nm_update_again_subnets_flow_rule_vrf.current.flowRuleAttributeList.0.subnet == "10.10.1.0/24"
+      - nm_update_again_subnets_flow_rule_vrf.current.flowRuleAttributeList.1.subnet == "10.10.2.0/24"
+      - nm_delete_subnets_flow_rule_vrf is changed
+      - nm_delete_subnets_flow_rule_vrf.data.flowRuleAttributeList.0.subnet == "10.10.1.0/24"
+      - nm_delete_subnets_flow_rule_vrf.data.flowRuleAttributeList.0.operation == "DELETE"
+      - nm_delete_subnets_flow_rule_vrf.data.flowRuleAttributeList.1.subnet == "10.10.2.0/24"
+      - nm_delete_subnets_flow_rule_vrf.data.flowRuleAttributeList.1.operation == "DELETE"
+      - nm_delete_subnets_flow_rule_vrf.current.name == "ansible_test"
+      - nm_delete_subnets_flow_rule_vrf.current.tenant == "test"
+      - nm_delete_subnets_flow_rule_vrf.current.vrf == "test"
+      - nm_delete_subnets_flow_rule_vrf.current.flowRuleAttributeList == []
+
+# QUEY TASKS
```

Comment:
Wouldn't it be better to put query tasks before deletion.

Reply:
You're right. It's probably better to put query tasks before any updates.